File: web/src/main/java/no/nav/modiapersonoversikt/service/organisasjonenhet/kontaktinformasjon/domain/Publikumsmottak.java
package no.nav.modiapersonoversikt.service.organisasjonenhet.kontaktinformasjon.domain;

public class Publikumsmottak {

    private Gateadresse besoeksadresse;
    private Apningstider apningstider;

    public Gateadresse getBesoeksadresse() {
        return besoeksadresse;
    }

    public Publikumsmottak withBesoeksadresse(Gateadresse besoeksadresse) {
        this.besoeksadresse = besoeksadresse;
        return this;
    }

    public Apningstider getApningstider() {
        return apningstider;
    }

    public Publikumsmottak withApningstider(Apningstider apningstider) {
        this.apningstider = apningstider;
        return this;
    }
}

File: web/src/main/java/no/nav/sbl/dialogarena/modiabrukerdialog/web/rest/enhet/model/Publikumsmottak.kt
package no.nav.sbl.dialogarena.modiabrukerdialog.web.rest.enhet.model

import no.nav.sbl.dialogarena.modiabrukerdialog.consumer.service.organisasjonenhet.kontaktinformasjon.domain.Publikumsmottak

class Publikumsmottak {
    var besoksadresse: Gateadresse
    var apningstider: List<Apningstid>

    constructor(publikumsmottak: Publikumsmottak) {
        this.besoksadresse = Gateadresse(
            publikumsmottak.besoeksadresse?.gatenavn,
            publikumsmottak.besoeksadresse?.husnummer,
            publikumsmottak.besoeksadresse?.husbokstav,
            publikumsmottak.besoeksadresse?.postnummer,
            publikumsmottak.besoeksadresse?.poststed
        )
        this.apningstider = publikumsmottak.apningstider.apningstider.map {
            Apningstid(
                it.ukedag.name,
                Klokkeslett(it.apentFra?.time, it.apentFra?.minutt, it.apentFra?.sekund),
                Klokkeslett(it.apentTil?.time, it.apentTil?.minutt, it.apentTil?.sekund)
            )
        }
    }
}
